recover.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"net/http"
+	"runtime"
+)
+
+// A recoverMux recovers from panics during an http.Handler, sending an error
+// response to a client in the event a panic occurs.
+type recoverMux struct {
+	mux *http.ServeMux
+
+	DumpStack bool // set to true to dump the stack to the client on a panic
+}
+
+// NewRecoverMux returns a new recoverMux, creating and wrapping a new
+// http.ServeMux.
+func newRecoverMux() *recoverMux {
```

Comment:
`nit` unless this is an internal only mux, you should probably expose the New func